### PR TITLE
[C] fix decode funtion

### PR DESCRIPTION
**Pull request by zhemant**
This PR includes:
- correct the function names to correct one as defined in model-body
- reformat base64decode function to avoid unnecessary malloc and assigning of the wrong length.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [ ] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by zhemant** [comment]
@wing328 @ityuhui  Please check the PR and let me know if it is okay

File: modules/openapi-generator/src/main/resources/C-libcurl/apiClient.c.mustache [hunk 1]
```diff
@@ -606,7 +606,7 @@ char *sbi_base64decode (const void *b64_decode_this, int decode_this_many_bytes)
     BIO_write(mem_bio, b64_decode_this, decode_this_many_bytes); //Base64 data saved in source.
     BIO_push(b64_bio, mem_bio);          //Link the BIOs by creating a filter-source BIO chain.
     BIO_set_flags(b64_bio, BIO_FLAGS_BASE64_NO_NL);          //Don't require trailing newlines.
-    int decoded_byte_index = 0;   //Index where the next base64_decoded byte should be written.
+    decoded_byte_index = 0;   //Index where the next base64_decoded byte should be written.
```
Comment (ityuhui) [inline modules/openapi-generator/src/main/resources/C-libcurl/apiClient.c.mustache]: I think 
```c
decoded_byte_index
```
shoud change to  
```
*decoded_byte_index
```
in the function "base64decode"

Comment (zhemant) [reply modules/openapi-generator/src/main/resources/C-libcurl/apiClient.c.mustache]: Thank @ityuhui , I changed the assigning to use local variable and then assign it to pointer as ++ is not working on pointer variables.

**Comment by ityuhui** [comment]
LGTM

**Comment by zhemant** [comment]
@wing328 This PR is ready for merge.

**Comment by wing328** [comment]
CI failures already fixed in the master.

**Comment by zhemant** [comment]
Unfortunately yes. I forgot there is a declaration in apiClient.h

I will fix it. I dont know if a merged PR can be merged again or should I just create a new PR with this? Any suggestion @ityuhui ?

**Comment by ityuhui** [comment]
@zhemant  
I don't know. Maybe needs a new PR ...

**Pull request merged**